explorer/src/components/Nav.tsx

```diff
@@ -1,138 +1,276 @@
   },
   {
-    label: 'Network Components',
-    route: '/network-components',
-    Icon: <GroupWork style={{ color: '#F2F2F2' }} />,
+    url: '/network-components',
+    title: 'Network Components',
+    Icon: <ConnectedTv />,
+    nested: [
+      {
+        url: '/network-components/mixnodes',
+        title: 'Mixnodes',
+        Icon: <TrafficSharp />,
+      },
+      {
+        url: '/network-components/gateways',
+        title: 'Gateways',
+        Icon: <GarageTwoTone />,
+      },
+    ],
   },
   {
-    label: 'Node Map',
-    route: '/nodemap',
-    Icon: <PinDrop style={{ color: '#F2F2F2' }} />,
+    url: '/nodemap',
+    title: 'Nodemap',
+    Icon: <Pin />,
   },
 ];
-export const Nav: React.FC = () => {
-  const [sidebar, setSidebar] = React.useState<boolean>(false);
-  const classes: ClassNameMap = useStyles();
-  const location = useLocation();
-  const showSidebar = () => setSidebar(!sidebar);
 
+const ExpandableButton: React.FC<navOptionType> = ({
+  nested,
+  title,
+  Icon,
+  url,
+}) => {
+  const [open, toggle] = React.useState(false);
+  const handleClick = () => toggle(!open);
+
+  if (!nested)
+    return (
+      <ListItem disableGutters component={Link} to={url}>
+        <ListItemButton
+          sx={{
+            color: (theme) =>
+              theme.palette.mode === 'light' ? '#000' : '#fff',
+          }}
+        >
+          <ListItemIcon>{Icon}</ListItemIcon>
+          <ListItemText
+            primary={title}
+            sx={{
+              color: (theme) =>
+                theme.palette.mode === 'light' ? '#000' : '#fff',
+            }}
+          />
+        </ListItemButton>
+      </ListItem>
+    );
   return (
     <>
-      <div
-        className={classes.navBar}
-        style={sidebar ? { width: 290 } : { width: 80 }}
-      >
-        <List>
+      <ListItem disableGutters>
+        <ListItemButton
+          onClick={handleClick}
+          sx={{
+            color: (theme) =>
+              theme.palette.mode === 'light' ? '#000' : '#fff',
+          }}
+        >
+          <ListItemIcon>{Icon}</ListItemIcon>
+          <ListItemText primary={title} />
+          {open ? <ExpandLess /> : <ExpandMore />}
+        </ListItemButton>
+      </ListItem>
+      {open &&
+        nested.map((each: navOptionType) => (
           <ListItem
-            button
-            className={classes.navListItem}
-            onClick={showSidebar}
-            style={{ height: 72 }}
+            disableGutters
+            key={each.url}
+            component={Link}
+            to={each.url}
           >
-            <ListItemIcon>
-              {sidebar ? (
-                <Close className={classes.hamburgerIcon} />
-              ) : (
-                <MenuIcon className={classes.hamburgerIcon} />
-              )}
-            </ListItemIcon>
+            <ListItemButton
+              sx={{
+                color: (theme) =>
+                  theme.palette.mode === 'light' ? '#000' : '#fff',
```

Comment:
The theme has a text colour in the palette that responds to the mode:

```suggestion
                color: 'text.primary',
```

```

![image](https://user-images.githubusercontent.com/14054343/134171306-b426a56f-50b1-4b21-82ca-972d4a11e810.png)

![image](https://user-images.githubusercontent.com/14054343/134171337-647e7eeb-a0e9-4309-b90a-2bd8931cf697.png)